### ISSUE TITLE
Rename English variants native name

### ISSUE THREAD

**Issue by dd32** [GlotPress/GlotPress-WP#1533]
The English and native name of the English variants is currently `English ($country)`, this is problematic in that it doesn't fit how the language is referred to within those locales. 

For example, `en_AU` should be referred to as `Australian English`, but likewise for en_NZ, en_CA, en_ZA, and en_GB.
Potentially, that should even be the "English" variant name (Which for the purpose of the GlotPress, is the en_US 'translation') for en_AU, en_GB, en_CA, en_NZ, and en_ZA. en_US would continue to have no `American` specifier, and en_GB would be forced to always have `British` in there (Even though, I suspect that within UK, it would just be referred to as English).

Wikipedia has a decent writeup of the dialects and their names:
https://en.wikipedia.org/wiki/List_of_dialects_of_English

I'm not 100% sure this issue is accurate, in that, I suspect it could be argued that WordPress.org is using `GP_Locale::$native_name` incorrectly as seen via https://meta.trac.wordpress.org/ticket/6600.

**Comment by dd32** [comment]
The other problem is that `en_GB` is currently labeled as `English (UK)` which is much more inclusive of all UK countries than `British English` is; But we also don't have a `Scottish English`, `Irish English`, or `Welsh English` variants (and probably don't need it).

UK vs GB is a bit confusing for the rest of the world, but to make it worse, UK doesn't include the Republic of Ireland + Isle of Man + Channel Islands AFAIK? making even `UK` not be technically correct either I think?
 - `Great Britain` (Countries on the island England exists on)
 - `United Kingdom` (The above + Northern Ireland)
 - `British Isles` (The above + Republic of Ireland + Isle of Man + Channel Islands)

_edit: Now that I've posted this, I realise that 'British English' could easily be suggested to be named after the British Isles, rather than Great Britain, despite the language code, I'll reference this from [Wikipedia on British English](https://en.wikipedia.org/wiki/British_English):_
> [Tom McArthur](https://en.wikipedia.org/wiki/Tom_McArthur_(linguist)) in the Oxford Guide to World English acknowledges that [British](https://en.wikipedia.org/wiki/British_people) English shares "all the ambiguities and tensions [with] the word 'British' and as a result can be used and interpreted in two ways, more broadly or more narrowly, within a range of blurring and ambiguity.

_I'll suffix this with a warning: I'm not from that part of the world, I may have inadvertently got the geographics or politics around names wrong_

**Comment by webaware** [comment]
I've not been contributing that much lately, but when I do I like to help complete a en_* and export it for import into en_{AU|NZ|CA} where I can localise. For example, en_AU and en_NZ are mostly the same except for greetings etc. (G'day vs Kia ora, etc.)

Sorting the projects by name helps me easily find the different English variants to compare and then do the export/import dance, because they all start off with "English (*)". Changing their names will break that. Maybe that's just my bad luck though.

Thumbs up for English (Scots), ah cannae wait!

**Comment by dd32** [comment]
> Sorting the projects by name helps me easily find the different English variants to compare 

This is a good reason potentially to keep the `english_name` field the same as current, which would keep with the other names.

Using the Chinese variants as an example:
|wp_locale|english_Name|native_name|Google translate|
|---|---|---|---|
|zh_CN|Chinese (China)|简体中文|Simplified Chinese|
|zh_HK|Chinese (Hong Kong)|香港中文|Hong Kong Chinese|
|zh_SG|Chinese (Singapore)|中文|Chinese|
|zh_TW|Chinese (Taiwan)|繁體中文|traditional Chinese|

(I'm sure Google Translate has made that a mess, but i've included it for us English speakers to easier differentiate the localised name)

**Comment by webaware** [comment]
Yup; see also German, Spanish, Portuguese, Norwegian, etc. Changing the `native_name` should be fine, but not the name listed in the sort column. For my purposes at least, but I'm pretty sure translators of other languages do similar (at least to break the back of large projects).

**Comment by ocean90** [comment]
I agree that using the `native_name` field as provided by `GP_Locales` isn't a good fit for the language banner. But I don't think we should change the current format which is mostly `$language ($country)`. By default the value is also used for the language dropdown in GlotPress but also the language selector in WordPress. Having a consistent prefix - the language - helps to find the language a bit easier and would be consistent with other language selectors:

WordPress:
<img width="336" alt="Bildschirm­foto 2022-11-28 um 09 27 31" src="https://user-images.githubusercontent.com/617637/204229547-e2b16699-2aaa-4c8b-a16c-e2e5d09ae10a.png">

Chrome:
<img width="555" alt="Bildschirm­foto 2022-11-28 um 09 21 08" src="https://user-images.githubusercontent.com/617637/204229827-91762fec-c5c9-46ea-9a2a-31856d86710c.png">

macOS:
<img width="533" alt="Bildschirm­foto 2022-11-28 um 09 28 10" src="https://user-images.githubusercontent.com/617637/204229861-a073fa73-2223-4f54-b332-e9989d56d52f.png">

For WordPress.org, maybe the language name should be a translatable string so that each locale can choose the best name for the use in the language banner?

**Comment by ntwb** [comment]
> For WordPress.org, maybe the language name should be a translatable string so that each locale can choose the best name for the use in the language banner?

Sounds good to me, solves the issue and doesn't break _other things_
